tools/xabuild/XABuildPaths.cs

```diff
@@ -130,7 +130,8 @@ public XABuildPaths ()
 				MSBuildConfig            = Path.Combine (MSBuildBin, "MSBuild.exe.config");
 				DotNetSdkPath            = FindLatestDotNetSdk (Path.Combine (Environment.GetEnvironmentVariable ("ProgramW6432"), "dotnet", "sdk"));
 				MSBuildSdksPath          = DotNetSdkPath ?? Path.Combine (MSBuildPath, "Sdks");
-				ProjectImportSearchPaths = new [] { MSBuildPath, "$(MSBuildProgramFiles32)\\MSBuild" };
+				MSBuildExtensionsPath    = Path.Combine (VsInstallRoot, "MSBuild");
```

Comment:
Umm.. on windows, XA would be installed to this location, correct? If yes, then it means that system XA would get priority :/

Assuming that for now, the reason we changed the order of lookup was (from the commit message):

```
Earlier xabuild was setting:

- MSBuildExtensionsPath: to the in-tree path with XA tasks/targets
- And setting the fallback search paths to the default
MSBuildExtensionsPath from system mono (and System `External/xbuild`
folder on macOS).

But this broke the test case `BuildBasicApplicationWithNuGetPackageConflicts` because [1]
`$(MSBuildExtensionsPath)\Microsoft\Microsoft.NET.Build.Extensions\Microsoft.NET.Build.Extensions.targets`
could not be imported from the in-tree path (which has only XA!). And
since this import was through the property:

`<MicrosoftNETBuildExtensionsTargets Condition="'$(MicrosoftNETBuildExtensionsTargets)' == ''">$(MSBuildExtensionsPath)\Microsoft\Microsoft.NET.Build.Extensions\Microsoft.NET.Build.Extensions.targets</MicrosoftNETBuildExtensionsTargets>`

.. the fallback paths for `$(MSBuildExtensionsPath)` weren't used.
```

So, we want to use the in-tree path as `$(MSBuildExtensionsPath)`, but we need that to have `Microsoft\Microsoft.NET.Build.Extension` (and maybe some other stuff?). Should we be duplicating the `MSBuildExtensionsPath` contents to the in-tree path?

Reply:
I don't have a VS with XA installed at the moment, but I think your concern is warranted. 

> Should we be duplicating the `MSBuildExtensionsPath` contents to the in-tree path?

In the MSBuild repo we handle a similar problem by copying all of the installed stuff to our folder, then overlaying it with the new bits, so yeah, I think that's a reasonable option..

Reply:
Ok, we happen to be doing the same thing, but with symlinks.

Once we get https://github.com/xamarin/xamarin-android/pull/2499 merged, I think we can rebase this one and everything will work.